src/common/utils/http.ts

```diff
@@ -0,0 +1,48 @@
+// ==LICENSE-BEGIN==
+// Copyright 2017 European Digital Reading Lab. All rights reserved.
+// Licensed to the Readium Foundation under one or more contributor license agreements.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file exposed on Github (readium) in the project repository.
+// ==LICENSE-END==
+
+import { container } from "readium-desktop/main/di";
+import { RootState } from "readium-desktop/main/redux/states";
+import { Store } from "redux";
+import * as request from "request";
+import { promisify } from "util";
+
+export async function httpGet(url: string, options?: request.CoreOptions): Promise<string> {
+
+    const store = container.get("store") as Store<RootState>;
+    const locale = store.getState().i18n.locale;
+    const headers = Object.assign(
+        {},
+        options.headers,
```

Comment:
`options` can be `undefined` (`options?: request.CoreOptions`).

`options.headers` can be `undefined` / `null`, but `Object.assign()` gracefully ignores:
"Note that Object.assign() does not throw on null or undefined source values."
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign

Reply:
done